code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -392,3 +392,87 @@
 
 /obj/item/weldingtool/robotic
 	icon = 'icons/obj/robot_items.dmi'
+
+/obj/item/inductive_charger
+	name = "inductive charger"
+	desc = "A phoron-enhanced induction charger hooked up to its attached stationbound's internal cell."
+	desc_fluff = "Harnessing the energy potential found in phoron structures, Nanotrasen engineers have created a portable device capable of highly efficient wireless charging. The expense and limit of energy output of using this method of charging prevents it from being used on a large scale, being far outclassed by Phoron-Supermatter charging systems."
+	desc_info = "Click on an adjacent object that contains or is a power cell to attempt to find and charge it. After a successful charge, the inductive charger recharge in a few minutes. The amount transfered can be adjusted by alt clicking it."
+	icon = 'icons/obj/robot_items.dmi'
+	icon_state = "inductive_charger"
+	var/ready_to_use = TRUE
+	var/recharge_time = 300
+	var/transfer_rate = 5000
+	maptext_x = 3
+	maptext_y = 2
+
+/obj/item/inductive_charger/Initialize()
+	. = ..()
+	maptext = "<span style=\"font-family: 'Small Fonts'; -dm-text-outline: 1 black; font-size: 7px;\">Ready</span>"
+
+/obj/item/inductive_charger/AltClick(mob/user)
+	. = ..()
+	var/set_rate = input(user, "How much do you want to transfer per use? (Limit: 1 - 5000)", "Induction Transfer Rate", 5000) as num
+	if(set_rate > 5000)
+		set_rate = 5000
+	else if(set_rate < 1)
+		set_rate = 1
+	transfer_rate = set_rate
+	to_chat(user, SPAN_NOTICE("You set the transfer rate of \the [src] to [transfer_rate]."))
```

Comment:
You should show how much energy is transferred to the target and how much energy is used to do that.